api/src/main/java/com/flipkart/poseidon/api/APILegoSet.java

```diff
@@ -57,7 +58,9 @@ public void updateBuildables(Map<String, Buildable> buildableMap) {
     @Override
     public Buildable getBuildable(Request request) throws LegoSetException, ElementNotFoundException {
         PoseidonRequest poseidonRequest = (PoseidonRequest) request;
-        Buildable buildable = trie.get(getKeysForTrie(poseidonRequest.getUrl()));
+        String completeUrl = ApiHelper.getUrlWithHttpMethod(poseidonRequest.getUrl(), poseidonRequest.getHttpMethod());
+        Buildable buildable = trie.get(getKeysForTrie(completeUrl));
+        buildable = (buildable == null) ? trie.get(getKeysForTrie(poseidonRequest.getUrl())) : buildable;
```

Comment:
Would be more readable to have
if (buildable == null) {
   buildable = trie.get(getKeysForTrie(poseidonRequest.getUrl());
}
Also avoids self assignment